odoo/addons/base/models/ir_property.py

```diff
@@ -24,13 +24,13 @@
     'boolean': bool,
     'integer': lambda val: val or False,
     'float': lambda val: val or False,
-    'char': lambda val: val or False,
-    'text': lambda val: val or False,
+    'char': lambda val: val if val is not None else False,
+    'text': lambda val: val if val is not None else False,
     'selection': lambda val: val or False,
     'binary': lambda val: val or False,
     'date': lambda val: val.date() if val else False,
     'datetime': lambda val: val or False,
-    'html': lambda val: val or False,
+    'html': lambda val: val if val is not None else False,
```

Comment:
I think this is a great idea, and according to my test it works fine.
I noticed that the `_String` version of `convert_to_record` does some weird stuff if the field is translatable, so this might be something we need to look out for if we want to have translatable company-dependent fields (Antony was mentioning we could do this if people complain about the account names not being translatable anymore)

Reply:
Yeah, let's assume there are no translatable company dependent fields for now; the ORM will take care of that (either supporting it with the new implementation directly, either adding support here if they can't finish the new implementation)